src/js/MetricsServiceClient.js

```diff
@@ -1,4 +1,29 @@
 import config from "../../config.json";
 import {MetricServicePromiseClient} from "./github.com/dappley/go-dappley/rpc/pb/rpc_grpc_web_pb";
-export const MetricsServiceClient = new MetricServicePromiseClient(`http://${config.host}:${config.port}`, null, null);
+import {MetricsServiceRequest} from "./github.com/dappley/go-dappley/rpc/pb/rpc_pb";
+
 export * from "./github.com/dappley/go-dappley/rpc/pb/rpc_pb";
+
+class _MetricServiceClient {
+    constructor() {
+        this.client = new MetricServicePromiseClient(`http://${config.host}:${config.port}`, null, null);
+    }
+
+    static getMetadata() {
+        return {Authorization: `Bearer ${sessionStorage.getItem("tkn")}`};
```

Comment:
Should requests contain `access-control-allow-origin` header? I tried deploying with IP and runs without dockers (except grpc-proxy) and I am still getting CORS problem

Reply:
Isn't `access-control-allow-origin` a response header, so it has to be sent server side? I think it would be completely normal to see such responses if one of the services were not running as they may not respond appropriately but I've never encountered the issue otherwise. To the best of my knowledge either grpc-proxy or auth-server are not accepting your request so https://github.com/dappley/dappley-stats/blob/0e2bc68adacd79d8b43fc964f3086b63a106f9ca/cm/templates/docker/grpc-proxy/envoy.yaml.mustache#L32-L41
or 
https://github.com/dappley/dappley-stats/blob/0e2bc68adacd79d8b43fc964f3086b63a106f9ca/auth_server/main.ts#L16.
Will need to be revised. You might be able to narrow done the search by determining which server is not replying with the correct headers from the port number?

Reply:
@harmangakhal My bad, I meant to put it in the response header. I was able to get to the loading component so authentication was working. I think it is grpc-proxy that causes the issue.

Reply:
You can open a bash shell in the container using, `docker exec -it auth-server bash` and look at the `/tmp/envoy.log`, it logs each request/response.

Reply:
Thanks for the advice. I figured it out. `auth-server` has to be run with docker so `grpc-proxy` can locate it.

Reply:
This seems promising in forwarding the headers, https://www.envoyproxy.io/docs/envoy/v1.10.0/api-v2/config/filter/http/ext_authz/v2/ext_authz.proto#envoy-api-msg-config-filter-http-ext-authz-v2-authorizationresponse; but as long as the request is rejected, I don't think it matters whether the client receives `access-control-allow-origin`.